### PR TITLE
Press-Schechter Mass Function

### PR DESCRIPTION
**Pull request by Lucia-Fonseca**
## Description
The halo mass function from the Press-Schechter formalism. This pull-request addresses issue #12.

## References
* Press, W. H. and Schechter, P., APJ, (1974).

## Checklist
- [X] Follow the [Contributor Guidelines](https://github.com/skypyproject/skypy/blob/develop/CONTRIBUTING.md)
- [X] Write unit tests
- [X] Write documentation strings
- [X] Assign someone from your working team to review this pull request
- [X] Assign someone from the infrastructure team to review this pull request

## Additional
- It needs more tests (suggestions?)

### CODE REVIEW

File: skypy/halo/tests/test_mass.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+# def test_press_schechter():
+
+# Test that sampling corresponds to sampling from the right pdf.
+# For this, we sample an array of luminosities for redshift z = 1.0 and we
+# compare it to the corresponding cdf.
+
+# sample = mass.press_schechter(1.0, 1e9, size=1000)
+# p_value = scipy.stats.kstest(sample, calc_cdf)[1]
+# assert p_value >= 0.01
```
Comment (rrjbca) [inline skypy/halo/tests/test_mass.py:15]: I suggest a test similar to the kstest in test_herbel_luminosities comparing the Press-Schechter samples against the analytic cdf

**Comment by Lucia-Fonseca** [comment]
I am getting this error:
`assert nan > 0.01

skypy/halo/tests/test_mass.py:25: AssertionError`
The p value is `nan`. Any idea why? Am I using unrealistic values?

**Comment by rrjbca** [comment]
> I am getting this error:
> `assert nan > 0.01
> 
> skypy/halo/tests/test_mass.py:25: AssertionError`The p value is`nan`. Any idea why? Am I using unrealistic values?

I assume you are running the tests locally? The tests won't run here until #93 is merged, once that is done I'll have a look and see if it's working

File: skypy/halo/mass.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    ...                               x_max=1e2, resolution=1000)
+
+    References
+    ----------
+    .. [1] Press, W. H. and Schechter, P., APJ, (1974).
+    """
+
+    alpha = - 0.5 * (n + 9.0) / (n + 3.0)
+
+    x_sample = schechter(alpha, x_min, x_max, resolution=100, size=size)
```
Comment (rrjbca) [inline skypy/halo/mass.py:53]:
```suggestion
    x_sample = schechter(alpha, x_min, x_max, resolution=resolution, size=size)
```

**Pull request merged**